plugins/woocommerce-admin/client/marketing/overview-multichannel/LearnMarketing.tsx

```diff
@@ -0,0 +1,191 @@
+/**
+ * External dependencies
+ */
+import { useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { Pagination, EmptyContent } from '@woocommerce/components';
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import { CollapsibleCard, ReadBlogMessage } from '../components';
+import { STORE_KEY } from '../data/constants';
+import './LearnMarketing.scss';
+
+type Post = {
+	author_avatar: string;
+	author_name: string;
+	date: string;
+	image: string;
+	link: string;
+	title: string;
+};
+
+type PostTileProps = {
+	post: Post;
+};
+
+const PlaceholderPostTile = () => {
+	return (
+		<div className="woocommerce-marketing-learn-marketing-card__post">
+			<div className="woocommerce-marketing-learn-marketing-card__post-img woocommerce-marketing-learn-marketing-card__post-img--placeholder"></div>
+			<div className="woocommerce-marketing-learn-marketing-card__post-title woocommerce-marketing-learn-marketing-card__post-title--placeholder"></div>
+			<div className="woocommerce-marketing-learn-marketing-card__post-description woocommerce-marketing-learn-marketing-card__post-description--placeholder"></div>
+		</div>
+	);
+};
+
+const PostTile: React.FC< PostTileProps > = ( { post } ) => {
+	return (
+		<a
+			className="woocommerce-marketing-learn-marketing-card__post"
+			href={ post.link }
+			target="_blank"
+			rel="noopener noreferrer"
+			onClick={ () => {
+				recordEvent( 'marketing_knowledge_article', {
+					title: post.title,
+				} );
+			} }
+		>
+			<div className="woocommerce-marketing-learn-marketing-card__post-img">
+				{ post.image && <img src={ post.image } alt="" /> }
+			</div>
+			<div className="woocommerce-marketing-learn-marketing-card__post-title">
+				{ post.title }
+			</div>
+			<div className="woocommerce-marketing-learn-marketing-card__post-description">
+				{
+					// translators: %s: author's name.
+					sprintf( __( 'By %s', 'woocommerce' ), post.author_name )
+				}
+				{ post.author_avatar && (
+					<img
+						src={ post.author_avatar.replace( 's=96', 's=32' ) }
+						alt=""
+					/>
+				) }
+			</div>
+		</a>
+	);
+};
+
+const blogPostCategory = 'marketing';
+const perPage = 2;
+
+const LearnMarketing = () => {
```

Comment:
💅 Missed return type in this and other functions

Reply:
I remember in my past reading that it is better to rely on inferred return types. One benefit is that there would be less code clutter. Also, when we change the function implementation and the return type is changed, usually it would break the caller function too and we would get compile error. In other words, the benefits of specifying return types may not worth the effort, especially for functions used only within a small module like Marketing page and not used elsewhere.

Reply:
Interesting. Ok